api/include/opentelemetry/logs/logger.h

```diff
@@ -57,7 +56,6 @@ class Logger
    * in order to create a log record.
    */
   virtual void Log(Severity severity,
-                   nostd::string_view name,
```

Comment:
I may be selfish here for my internal customers, even though we agreed to make it a breaking change, do you think keep `name` as something macro-enabled or deprecated or add separate methods without `name`? If it is too much of a change, for now, we can keep it as it is, and I will raise a separate PR on top of this to add `name` as deprecated/macro-enabled.

Reply:
Please ignore this comment, going through the PR I think it would need some management effort to keep both the APIs available.

Reply:
Fine by me to leave the Logger APIs with name for one or two versions, but the new proto has removed name field of `LogRecord` , so we will just ignore the parameter and just leave some time for users to migrate codes.

Reply:
Yes, it would be awesome if we can keep the Logger APIs unchanged ( for the `name` parameter ) for some time so as to not break instrumentation library compilation :)

Reply:
APIs with `name` are restored.